mlir/lib/Dialect/MIOpen/utility/transformMapUtils.cpp

```diff
@@ -302,7 +352,10 @@ propagateVectorizationInfo(TransformMapAttr map,
         uint32_t upper, lower;
         std::tie(upper, lower) = data.value();
         if (input[upper].hasValue()) {
-          result[lower] = math_util::gcd(*input[upper], sliceEnd - sliceBegin);
+          int64_t maxLength =
```

Comment:
Slice is specified by the beginning and end indices

Reply:
Should you also gcd against begin length and end length? Otherwise the vectorization may cross the boundary of sliced chunk.

Reply:
Well, no, because the map is `(dX) -> (dX + begin)`

Shifting values doesn't affect chunk validity

Reply:
Does it matter if the begin happen to in middle of the vectorization length? For example, if we are to shift the element by 3, and the vectorization length is 10. It can still vectorize from the 4th elements going forward?

Reply:
Hm ... 

Considering [a, b, c, d] having been [x, a, b, c, d, y] in the underlying space, you can still vectorize by 4 - you'll be starting to read at `a` and can still go 4 (or 2) or so on at a time. You'll definitely run into `needsCoefficient` problems if you try to recombine this longer dimension with something else - same deal as padding, but the vectorization itself is safe.